packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -984,4 +989,56 @@ mod tests {
 
         let _bindings = contract.generate().unwrap();
     }
+
+    #[test]
+    fn error_on_empty_function_name() {
+        let contract = r#"
+        [
+            {
+                "type":"contract",
+                "inputs":[
+                    {
+                        "name":"arg1",
+                        "type":"u32"
+                    }
+                ],
+                "name":"",
+                "outputs":[]
+            }
+        ]
+        "#;
+
+        if let Some(result_error) = Abigen::new("test", contract).err() {
```

Comment:
Nice catch. Thanks!
I like your second approach. It is clear and produces nice error messages.

Reply:
What do you think about this?
```rust
  let result_error = Abigen::new("test", contract)?
      .generate()
      .err()
      .expect("Expected an error because the function name is empty");

  assert_eq!(
      result_error.to_string(),
      "Invalid data: Function name can not be empty"
  );
```
I think it is more concise and clearer.